src/NHibernate/Impl/MultipleQueriesCacheAssembler.cs

```diff
@@ -77,7 +77,7 @@ public object Assemble(object cached, ISessionImplementor session, object owner)
 			if (!queryParameters.ForceCacheRefresh)
 			{
 				IList list =
-					queryCache.Get(key, new ICacheAssembler[] {this}, queryParameters.NaturalKeyLookup, querySpaces, session);
+					queryCache.Get(key, queryParameters, new ICacheAssembler[] {this}, querySpaces, session);
```

Comment:
Multi-criteria/query do not handle readonly globally either (at least their `queryParameters` readonly setting is not set anywhere), so calling the new `Get` will not introduce a handling which was not here previously.